src/sisl/io/vasp/tests/test_chg.py

```diff
@@ -40,3 +37,36 @@ def test_graphene_chgcar_index_float(sisl_files):
     gridh = chgSileVASP(f).read_grid(index=[0.5])
 
     assert grid.grid.sum() / 2 == pytest.approx(gridh.grid.sum())
+
+
+@pytest.fixture(scope="module", params=["unpol", "pol", "soi"])
+def spin_conf(request):
+    return request.param
+
+
+def test_nitric_oxide_chg(sisl_files, spin_conf, chg_type, np_dtype):
+    f = sisl_files(_dir, f"nitric_oxide/{spin_conf}", chg_type + ".gz")
+    grid = chgSileVASP(f).read_grid(dtype=np_dtype)
+    geom = chgSileVASP(f).read_geometry()
+
+    assert grid.grid.sum() * grid.dvolume == pytest.approx(11)
+    assert geom == grid.geometry
+
+
+def test_nitric_oxide_pol(sisl_files, chg_type, np_dtype):
+    f = sisl_files(_dir, "nitric_oxide/pol", chg_type + ".gz")
+    grid = chgSileVASP(f).read_grid(1, dtype=np_dtype)
+    geom = chgSileVASP(f).read_geometry()
+
+    assert grid.grid.sum() * grid.dvolume == pytest.approx(1, rel=1e-3)
+    assert geom == grid.geometry
+
+
+def test_nitric_oxide_soi(sisl_files, chg_type, np_dtype):
+    f = sisl_files(_dir, "nitric_oxide/soi", chg_type + ".gz")
+    s = 0
+    for i in range(1, 4):
```

Comment:
for SOI, shouldn-t it have 4 indices? This only uses 3 of them?

Reply:
Correct, `index=0` is charge and the following three are x,y,z components (as far as I understand).

Reply:
ok, I see, so here you are checking spin. I'll have a look at this tomorrow. I am going to bed now! :bowing_man:

Reply:
According to https://www.vasp.at/wiki/index.php/CHGCAR

`pol`:
- `index=0` is charge density (spin up + spin down)
- `index=1` is magnetization (spin up - spin down)

`soi`:
- `index=0` is charge density
- `index=1` is magnetization density in $\sigma_1$ direction
- `index=2` is magnetization density in $\sigma_2$ direction
- `index=3` is magnetization density in $\sigma_3$ direction